core/iwasm/compilation/aot_emit_aot_file.c

```diff
@@ -1592,16 +1601,12 @@ get_name_section_size(AOTCompData *comp_data)
         return 0;
     }
 
-    if (!check_utf8_str(p, name_len)) {
-        aot_set_last_error("invalid UTF-8 encoding");
-        return 0;
-    }
-
+    /* "name" */
```

Comment:
Should we check whether `name_len == 4`? Suppose name_len == 3, the below memcmp may overread.
And it seems that loader also doesn't check it:
https://github.com/bytecodealliance/wasm-micro-runtime/blob/main/core/iwasm/interpreter/wasm_loader.c#L5058

Reply:
roll back following `EMIT_BUF()`